docs: smart transform supports Message type (#11306)
langflow-ai/langflow: docs/docs/Components/smart-transform.mdx

This component has been renamed multiple times.
Its previous names include **Lambda Filter** and **Smart Function**.

> The **Smart Transform** component uses an LLM to generate a Lambda function to filter or transform structured data or messages based on natural language instructions.

Reviewer: ```suggestion
The **Smart Transform** component uses an LLM and natural language instructions to generate a Lambda function that can filter or transform [`Data`](/data-types#data), [`DataFrame`](/data-types#dataframe), or [`Message`](/data-types#message) input.
```

| Name | Display Name | Info |
|------|--------------|------|

> | data | Data | Input parameter. The structured data or messages to filter or transform using a Lambda function. Accepts [`Data`](/data-types#data), [`DataFrame`](/data-types#dataframe), or [`Message`](/data-types#message) types. |

Reviewer: ```suggestion
| data | Data | Input parameter. The [`Data`](/data-types#data), [`DataFrame`](/data-types#dataframe), or [`Message`](/data-types#message) input to filter or transform using the generated Lambda function. |
```

| model | Language Model | Input parameter. Connect [`LanguageModel`](/data-types#languagemodel) output from a **Language Model** component. |
| filter_instruction | Instructions | Input parameter. The natural language instructions for how to filter or transform the data. The LLM uses these instructions to create a Lambda function. |
| sample_size | Sample Size | Input parameter. For large datasets, the number of characters to sample from the dataset head and tail. Only applied if the dataset meets or exceeds `max_size`. Default: `1000`. |
| max_size | Max Size | Input parameter. The number of characters for the dataset to be considered large, which triggers sampling by the `sample_size` value. Default: `30000`. |

## Smart transform output

The **Output** type should match the input type for reliable results.
For example, select **Message** output for **Message** input.

The **Smart Transform** component provides multiple output type options:

- **Data Output** returns the transformed result as a [`Data`](/data-types#data) object.
- **DataFrame Output** returns the transformed result as a [`DataFrame`](/data-types#dataframe) object.

> - **Message Output** returns the transformed result as a [`Message`](/data-types#message) object.

Reviewer: The output types are listed on line 17, so I think you could just add the first two lines of this section to the tip box ~line 24:

```
For the most reliable results, the **Smart Transform** component's output type must match the input type.
For example, select **Message** output for [`Message`](/data-types#message) input.
```

And then remove this section.
